### ISSUE TITLE
Mobile menu icon a11y

### ISSUE THREAD

**Issue by Amankumar29** [creativecommons/cccatalog-frontend#736]
## Describe the bug
when we use tab key in the mobile view the toggle icon is not properly working and the components inside it is not displaying i think it should be corrected for better accessibility.

## To Reproduce

1. Go to https://ccsearch.creativecommons.org/
2. Click on 'Tab key'
3. See error

## Expected behavior
when we go to toggle button it should automatically open and the components under it should be displayed,so the items can easily be selected.

## Additional context
if really its a bug,i would like to take this up.

**Comment by annatuma** [comment]
We are unable to replicate this. Please surface this in the cc-search channel on Slack and see if others are able to. Recording the screen when this happens would also be helpful. Please indicate what mobile device and browser you are using as well.

If more details aren't available, we should not move forward with this.

cc @akmadian

**Comment by akmadian** [comment]
I'm not sure why anyone would be using search on a mobile breakpoint, and using tab for navigation (maybe there's a use case that I'm unfamiliar with?), although that's not a reason to not implement a fix.

I think this may be to do with the nav hamburger, the links inside the menu (which is hidden) can be selected, but the hamburger does not expand when a link inside it is focused on. This creates an effect where the links can be focused on and selected, but not seen. 

@coderbabu10 Is this what you are referring to? If so, I think it would be a good idea to move forward with a fix for this where users would have to focus on and select the hamburger before the links exist in the DOM.

**Comment by coderbabu10** [comment]
as there are many people who can't afford the laptops and also some having some physical disorder so they can simply connect it with the keyboard and use it with the flow.

in mobile view, when we use the tab key i think it would be more appropriate that the toggle button should open automatically when tab key pointed towards it and then it should point the components inside it as it will be more appropriate from my side and will improve the accessibility,what you think? @akmadian @annatuma 
or we can do the changes that @akmadian has suggested that we can make the tab key to the toggle icon then press enter and get the components inside it.

**Comment by akmadian** [comment]
In general, it is bad for UX and accessibility for anything to happen that was not explicitly triggered by the user. So, I think that having the hamburger auto expand would be bad from a UX perspective.

I think it would be a good idea to move forward with a fix for this. Go ahead and work on it @coderbabu10!

**Comment by abhisheknaiidu** [comment]
@akmadian @kgodey I guess these were the issues that comes under improving accessibility, right? and it needs to get solved during the Summer of Code, right? Not now i guess

**Comment by zackkrida** [comment]
Hi all, the ideal behavior here is that the menu icon is tabbable, has a clear visual style when focused, and upon focus the user can use the 'Enter' key to open/close the menu. 

That is generally the expected behavior for keyboard and screen reader users.

I will leave this open for now, may close later based on the progress of our GSoC a11y efforts. Thanks all!

**Comment by zackkrida** [comment]
@abhisheknaiidu you're welcome to tackle this _or_ @AyanChoudhary can include it in his accessibility work. Either works!

**Comment by AyanChoudhary** [comment]
@zackkrida I guess we have covered this issue and still since it is not conventional for people to use tabs for navigating in a mobile device, it makes sense to close this issue. Over to you for further actions.
